### DISCUSSION TITLE
Making shrine easier to use like NextJS's Image optimizations

### DISCUSSION THREAD

**Discussion by gczh** [shrinerb/shrine#645] [Ideas]
When using NextJS, there's the nifty <Image /> component provided by the framework.

When using it, it does [Image Optimizations](https://nextjs.org/docs/pages/building-your-application/optimizing/images) out of the box and generates multiple variants of an image to be served for different screen sizes.

I'm wondering if there is any way for Shrine to provide a similar developer ergonomics such as this? I get that Shrine is fundamentally a file attachment toolkit. However, as it's so flexible, i believe it'll be interesting provide a wrapper around Shrine to provide the same image optimisation ergonomics NextJS provides. 

I'd be happy to work with anyone to write a plugin or something for this if they're able to guide me!

**Comment by janko** [comment]
Shrine is a backend library, and Ruby apps don't have any build step, so it's a different architecture than Next.js. Next.js probably does pre-processing of different variants during building, which is possible because it controls generated HTML. Shrine cannot do that, because Ruby/Rails only evaluates HTML templates at runtime.
